File: 투포인터/수들의합2.py
n, m = map(int, input().split())
arrays = list(map(int, input().split()))


start_index = 0
end_index = 0
current_sum = arrays[start_index]
answer = 0

while(True):
    if start_index >= n:
        break

    if arrays[end_index] >= m:
        if arrays[end_index] == m:
            answer += 1

        if end_index < n - 1:
            end_index += 1
            start_index = end_index
            current_sum = arrays[end_index]
            continue
        else:
            break

    if current_sum < m:
        end_index += 1
        if end_index >= n:
            break
        current_sum += arrays[end_index]
    elif current_sum > m:
        current_sum -= arrays[start_index]
        start_index += 1
    elif current_sum == m:
        answer += 1
        current_sum -= arrays[start_index]
        start_index += 1    

print(answer)

File: 투포인터/수들의합2.java
package 투포인터;

import java.util.Scanner;

class 수들의합2 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt(); // 배열 길이
        int m = sc.nextInt(); // 목표 값
        int[] arrays = new int[n];

        for (int i = 0; i < n; i++) {
            arrays[i] = sc.nextInt();
        }

        int startIndex = 0;
        int endIndex = 0;
        int currentSum = arrays[startIndex];
        int answer = 0;

        while (true) {
            if (startIndex >= n) {
                break;
            }

            if (arrays[endIndex] >= m) {
                if (arrays[endIndex] == m) {
                    answer++;
                }

                if (endIndex < n - 1) {
                    endIndex++;
                    startIndex = endIndex;
                    currentSum = arrays[endIndex];
                    continue;
                } else {
                    break;
                }
            }

            if (currentSum < m) {
                endIndex++;
                if (endIndex >= n) {
                    break;
                }
                currentSum += arrays[endIndex];
            } else if (currentSum > m) {
                currentSum -= arrays[startIndex];
                startIndex++;
            } else if (currentSum == m) {
                answer++;
                currentSum -= arrays[startIndex];
                startIndex++;
            }
        }

        System.out.println(answer);
        sc.close();
    }
}